libcentrifugo/engine/engineredis/hash_test.go

```diff
@@ -0,0 +1,16 @@
+package engineredis
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestHash(t *testing.T) {
+	chans := []string{"chan1", "chan2", "chan3", "chan4", "chan5", "chan6", "chan7", "chan8"}
+	for _, ch := range chans {
+		bucket := Hash(ch, 2)
+		assert.True(t, bucket >= 0)
+		assert.True(t, bucket < 2)
+	}
+}
```

Comment:
You could add a probabilistic test of the hashing here, it might not be a good candidate for CI as it would be not be reproducible, but it's hard to have confidence in implementation of something like a hash function unless there are code tests that at least sanity check that for some example(s) it produces roughly the expected distributions.

In the past I've written them where I've picked generous bounds like hash 1000 random keys into 5 buckets and assert that each bucket has between 150 - 250 items.

You could also assert that adding a bucket doesn't move more than 20% keys (plus some margin for error/random chance so say 30%). If you make it tolerant enough it's unlikely to fail while still sanity checking there is no dumb typo introduced in the algorithm that will badly skew it's performance in practice.

I've seen this before where someone didn't test probabilistic code because it was no-deterministic like this but then when I analysed it, they had an off-by-one error that had skewed group assignment in A/B tests for years by enough that it basically invalidated all the conclusions drawn from several years of running A/B tests :(